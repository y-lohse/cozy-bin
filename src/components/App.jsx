/* global cozy */
import React from 'react'
import Form from './Form'
import Link from './Link'

class App extends React.Component {
  state = {
    url: null
  }

  createBin = async data => {
    try {
      const doc = await cozy.client.data.create('io.cozy.bin', data)
      const { _id } = doc
      const sharingData = {
        data: {
          type: 'io.cozy.permissions',
          attributes: {
            source_id: 'io.cozy.apps/cozy-bin',
            permissions: {
              bin: {
                type: 'io.cozy.bin',
                verbs: ['GET'],
                values: [_id]
              }
            }
          }
        }
      }
      const resp = await cozy.client.fetchJSON('POST', '/permissions?codes=all', sharingData)
      const code = resp.attributes.codes.all
      const url = new URL(window.location)

      this.setState({
        url: `${url.origin}/bin?sharecode=${code}`
      })
    } catch (e) {
      console.warn(e)
    }
  }

  render () {
    const { url } = this.state

    return (
      <div className='app-wrapper c-layout'>
        <main className='app-content'>
          <div role="contentinfo">
            { url
              ? <Link url={url} />
              : <Form onSubmit={this.createBin} />
            }
          </div>
        </main>
      </div>
    )
  }
}

export default App
