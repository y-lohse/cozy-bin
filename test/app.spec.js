'use strict'

/* eslint-env jest */

import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import App from '../src/components/App'

Enzyme.configure({ adapter: new Adapter() })

describe('App component only', () => {
  it('should be mounted correctly', () => {
    const component = shallow(<App />).getElement()
    expect(component).toMatchSnapshot()
  })
})
