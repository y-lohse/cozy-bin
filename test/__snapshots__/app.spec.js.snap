// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component only should be mounted correctly 1`] = `
<HashRouter>
  <div
    className="app-wrapper o-layout--2panes"
  >
    <Wrapper />
    <main
      className="app-content"
    >
      <Switch>
        <Route
          component={[Function]}
          path="/viewhello1"
        />
        <Route
          component={[Function]}
          path="/viewhello2"
        />
        <Route
          component={[Function]}
          path="/viewhello3"
        />
        <Redirect
          from="/"
          push={false}
          to="/viewhello1"
        />
        <Redirect
          from="*"
          push={false}
          to="/viewhello1"
        />
      </Switch>
    </main>
  </div>
</HashRouter>
`;
